### ISSUE TITLE
fail to install collections: Symbol not found: _holes_clear

### ISSUE THREAD

**Issue by smartinsightsfromdata** [randy3k/collections#5]
I'm unable to upgrade to the latest release of collections.

This is my `sessioninfo()`

```
version 3.6.1 (2019-07-05)
Platform: x86_64-apple-darwin18.7.0 (64-bit)
Running under: macOS Catalina 10.15

Matrix products: default
BLAS:   /System/Library/Frameworks/Accelerate.framework/Versions/A/Frameworks/vecLib.framework/Versions/A/libBLAS.dylib
LAPACK: /usr/local/Cellar/openblas/0.3.7/lib/libopenblasp-r0.3.7.dylib

locale:
[1] en_GB.UTF-8/en_GB.UTF-8/en_GB.UTF-8/C/en_GB.UTF-8/en_GB.UTF-8

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

loaded via a namespace (and not attached):
[1] compiler_3.6.1 tools_3.6.1    packrat_0.5.0
```
This is what is happening:
```
install.packages("collections")
Installing package into ‘/Users/e/Library/R/userLibrary’
(as ‘lib’ is unspecified)
trying URL 'https://cran.rstudio.com/src/contrib/collections_0.2.1.tar.gz'
Content type 'application/x-gzip' length 45896 bytes (44 KB)
==================================================
downloaded 44 KB

* installing *source* package ‘collections’ ...
** package ‘collections’ successfully unpacked and MD5 sums checked
** using staged installation
** libs
clang -I"/usr/local/Cellar/r/3.6.1/lib/R/include" -DNDEBUG   -I/usr/local/opt/gettext/include -I/usr/local/opt/readline/include -I/usr/local/opt/icu4c/include -I/usr/local/include  -fPIC  -I/Library/Java/JavaVirtualMachines/jdk-8.0_172.jdk/Contents/Home/include -I/Library/Java/JavaVirtualMachines/jdk-8.0_172.jdk/Contents/Home/include/darwin  -c tommyds/tommy.c -o tommyds/tommy.o
clang -I"/usr/local/Cellar/r/3.6.1/lib/R/include" -DNDEBUG   -I/usr/local/opt/gettext/include -I/usr/local/opt/readline/include -I/usr/local/opt/icu4c/include -I/usr/local/include  -fPIC  -I/Library/Java/JavaVirtualMachines/jdk-8.0_172.jdk/Contents/Home/include -I/Library/Java/JavaVirtualMachines/jdk-8.0_172.jdk/Contents/Home/include/darwin  -c collections.c -o collections.o
clang -I"/usr/local/Cellar/r/3.6.1/lib/R/include" -DNDEBUG   -I/usr/local/opt/gettext/include -I/usr/local/opt/readline/include -I/usr/local/opt/icu4c/include -I/usr/local/include  -fPIC  -I/Library/Java/JavaVirtualMachines/jdk-8.0_172.jdk/Contents/Home/include -I/Library/Java/JavaVirtualMachines/jdk-8.0_172.jdk/Contents/Home/include/darwin  -c deque.c -o deque.o
clang -I"/usr/local/Cellar/r/3.6.1/lib/R/include" -DNDEBUG   -I/usr/local/opt/gettext/include -I/usr/local/opt/readline/include -I/usr/local/opt/icu4c/include -I/usr/local/include  -fPIC  -I/Library/Java/JavaVirtualMachines/jdk-8.0_172.jdk/Contents/Home/include -I/Library/Java/JavaVirtualMachines/jdk-8.0_172.jdk/Contents/Home/include/darwin  -c dict.c -o dict.o
clang -I"/usr/local/Cellar/r/3.6.1/lib/R/include" -DNDEBUG   -I/usr/local/opt/gettext/include -I/usr/local/opt/readline/include -I/usr/local/opt/icu4c/include -I/usr/local/include  -fPIC  -I/Library/Java/JavaVirtualMachines/jdk-8.0_172.jdk/Contents/Home/include -I/Library/Java/JavaVirtualMachines/jdk-8.0_172.jdk/Contents/Home/include/darwin  -c priority_queue.c -o priority_queue.o
clang -I"/usr/local/Cellar/r/3.6.1/lib/R/include" -DNDEBUG   -I/usr/local/opt/gettext/include -I/usr/local/opt/readline/include -I/usr/local/opt/icu4c/include -I/usr/local/include  -fPIC  -I/Library/Java/JavaVirtualMachines/jdk-8.0_172.jdk/Contents/Home/include -I/Library/Java/JavaVirtualMachines/jdk-8.0_172.jdk/Contents/Home/include/darwin  -c queue.c -o queue.o
clang -I"/usr/local/Cellar/r/3.6.1/lib/R/include" -DNDEBUG   -I/usr/local/opt/gettext/include -I/usr/local/opt/readline/include -I/usr/local/opt/icu4c/include -I/usr/local/include  -fPIC  -I/Library/Java/JavaVirtualMachines/jdk-8.0_172.jdk/Contents/Home/include -I/Library/Java/JavaVirtualMachines/jdk-8.0_172.jdk/Contents/Home/include/darwin  -c stack.c -o stack.o
clang -I"/usr/local/Cellar/r/3.6.1/lib/R/include" -DNDEBUG   -I/usr/local/opt/gettext/include -I/usr/local/opt/readline/include -I/usr/local/opt/icu4c/include -I/usr/local/include  -fPIC  -I/Library/Java/JavaVirtualMachines/jdk-8.0_172.jdk/Contents/Home/include -I/Library/Java/JavaVirtualMachines/jdk-8.0_172.jdk/Contents/Home/include/darwin  -c utils.c -o utils.o
clang -dynamiclib -Wl,-headerpad_max_install_names -undefined dynamic_lookup -single_module -multiply_defined suppress -L/usr/local/Cellar/r/3.6.1/lib/R/lib -L/usr/local/opt/openblas/lib -L/usr/local/opt/gettext/lib -L/usr/local/opt/readline/lib -L/usr/local/opt/icu4c/lib -L/usr/local/lib -o collections.so tommyds/tommy.o collections.o deque.o dict.o priority_queue.o queue.o stack.o utils.o -L/usr/local/Cellar/r/3.6.1/lib/R/lib -lR -lintl -Wl,-framework -Wl,CoreFoundation
installing to /Users/e/Library/R/userLibrary/00LOCK-collections/00new/collections/libs
** R
** byte-compile and prepare package for lazy loading
** help
*** installing help indices
** building package indices
** testing if installed package can be loaded from temporary location
Error: package or namespace load failed for ‘collections’ in dyn.load(file, DLLpath = DLLpath, ...):
 unable to load shared object '/Users/e/Library/R/userLibrary/00LOCK-collections/00new/collections/libs/collections.so':
  dlopen(/Users/e/Library/R/userLibrary/00LOCK-collections/00new/collections/libs/collections.so, 6): Symbol not found: _holes_clear
  Referenced from: /Users/e/Library/R/userLibrary/00LOCK-collections/00new/collections/libs/collections.so
  Expected in: flat namespace
 in /Users/e/Library/R/userLibrary/00LOCK-collections/00new/collections/libs/collections.so
Error: loading failed
Execution halted
ERROR: loading failed
* removing ‘/Users/e/Library/R/userLibrary/collections’
* restoring previous ‘/Users/e/Library/R/userLibrary/collections’
Warning in install.packages :
  installation of package ‘collections’ had non-zero exit status

The downloaded source packages are in
	‘/private/var/folders/9v/251f7k_x6hn8t9wsz8v21rgh0000gn/T/RtmppoTOOx/downloaded_packages’
```

**Comment by randy3k** [comment]
To be honest, I am not sure what caused it. It’s complaining that a symbol cannot be found but it’s actually already defined. I am pretty sure it’s not an issue specific to collections. 

Perhaps it’s due to your recent update of the macOS and you are mixing files compiled will different versions of tool chain. Check if you have the latest version of Xcode too.

**Comment by randy3k** [comment]
I spoke too soon and it’s actually a collections issue. We didn't put `static` in those `inline` functions. (see https://stackoverflow.com/questions/10243018/inline-function-undefined-symbols-error)

A workaround for now is to install the prebuilt macOS binary or to compile the c files with `-std=gnu89` or `-O2`.

**Comment by randy3k** [comment]
14e3f709c3ebc102d7e5205d33495aa57b76a9a7 should fix it.

**Comment by randy3k** [comment]
By the way, R's default CFLAGS should be `-Wall -g -O2`, did you change it anyway? I recommend you at least keep the optimization flag `-O2`.

**Comment by smartinsightsfromdata** [comment]
@randy3k Thanks. Tested with `devtools::install_github("randy3k/collections")` and works.

I normally install R with `sethrfore/homebrew-r-srf` & I do not normally set renviron files. So not sure where the flags issue is coming from.

After downloading Catalina I've installed the latest version of command line tools: I do not use Xcode.

Please let me know if you have any suggestion to improve my tools chain.

**Comment by randy3k** [comment]
Did you create the file `.R/Makevars` and define `CFLAGS` or `PKG_CFLAGS`? 
Judged from your output, you might have defined something for java 
```
-I/Library/Java/JavaVirtualMachines/jdk-8.0_172.jdk/Contents/Home/include -I/Library/Java/JavaVirtualMachines/jdk-8.0_172.jdk/Contents/Home/include/darwin`
```


I tried the regular homebrew R, it uses `-g -O2`

```
clang -I"/usr/local/Cellar/r/3.6.1_1/lib/R/include" -DNDEBUG   -I/opt/X11/include -I/usr/local/opt/gettext/include  -fPIC  -g -O2  -c tommyds/tommy.c -o tommyds/tommy.o
clang -I"/usr/local/Cellar/r/3.6.1_1/lib/R/include" -DNDEBUG   -I/opt/X11/include -I/usr/local/opt/gettext/include  -fPIC  -g -O2  -c collections.c -o collections.o
```

**Comment by smartinsightsfromdata** [comment]
I think it comes from my `Makeconf` file:

```
# etc/Makeconf.  Generated from Makeconf.in by configure.
#
# ${R_HOME}/etc/Makeconf
#
# R was configured using the following call
# (not including env. vars and site configuration)
# configure  '--prefix=/usr/local/Cellar/r/3.6.1' '--enable-memory-profiling' '--with-x' '--with-aqua' '--with-lapack' '--enable-R-shlib' 'SED=/usr/bin/sed' '--with-tcltk' '--with-tcl-config=/System/Library/Frameworks/Tcl.framework/tclConfig.sh' '--with-tk-config=/System/Library/Frameworks/Tk.framework/tkConfig.sh' '--with-blas=-L/usr/local/opt/openblas/lib -lopenblas' '--enable-java' '--with-cairo' 'JAVA_HOME=/Library/Java/JavaVirtualMachines/jdk-8.0_172.jdk/Contents/Home' 'PKG_CONFIG_PATH=/usr/local/opt/isl/lib/pkgconfig:/usr/local/opt/mpfr/lib/pkgconfig:/usr/local/opt/jpeg/lib/pkgconfig:/usr/local/opt/libpng/lib/pkgconfig:/usr/local/opt/pcre/lib/pkgconfig:/usr/local/opt/readline/lib/pkgconfig:/usr/local/opt/xz/lib/pkgconfig:/usr/local/opt/openblas/lib/pkgconfig:/usr/local/opt/freetype/lib/pkgconfig:/usr/local/opt/fontconfig/lib/pkgconfig:/usr/local/opt/libffi/lib/pkgconfig:/usr/local/opt/openssl@1.1/lib/pkgconfig:/usr/local/opt/sqlite/lib/pkgconfig:/usr/local/opt/python/lib/pkgconfig:/usr/local/opt/glib/lib/pkgconfig:/usr/local/opt/lzo/lib/pkgconfig:/usr/local/opt/pixman/lib/pkgconfig:/usr/local/opt/cairo/lib/pkgconfig:/usr/local/opt/icu4c/lib/pkgconfig' 'PKG_CONFIG_LIBDIR=/usr/lib/pkgconfig:/usr/local/Homebrew/Library/Homebrew/os/mac/pkgconfig/10.14:/opt/X11/lib/pkgconfig:/opt/X11/share/pkgconfig' 'CC=clang' 'CFLAGS=-I/Library/Java/JavaVirtualMachines/jdk-8.0_172.jdk/Contents/Home/include -I/Library/Java/JavaVirtualMachines/jdk-8.0_172.jdk/Contents/Home/include/darwin' 'LDFLAGS=-L/usr/local/opt/openblas/lib -L/usr/local/opt/gettext/lib -L/usr/local/opt/readline/lib -L/usr/local/opt/icu4c/lib' 'CPPFLAGS=-I/usr/local/opt/gettext/include -I/usr/local/opt/readline/include -I/usr/local/opt/icu4c/include' 'CXX=clang++' 'CXXFLAGS=-I/Library/Java/JavaVirtualMachines/jdk-8.0_172.jdk/Contents/Home/include -I/Library/Java/JavaVirtualMachines/jdk-8.0_172.jdk/Contents/Home/include/darwin' 'OBJC=clang' 'OBJCFLAGS=-I/Library/Java/JavaVirtualMachines/jdk-8.0_172.jdk/Contents/Home/include -I/Library/Java/JavaVirtualMachines/jdk-8.0_172.jdk/Contents/Home/include/darwin'

## This fails if it contains spaces, or if it is quoted
include $(R_SHARE_DIR)/make/vars.mk

AR = ar
BLAS_LIBS = -L/usr/local/opt/openblas/lib -lopenblas
C_VISIBILITY = 
CC = clang
CFLAGS = -I/Library/Java/JavaVirtualMachines/jdk-8.0_172.jdk/Contents/Home/include -I/Library/Java/JavaVirtualMachines/jdk-8.0_172.jdk/Contents/Home/include/darwin $(LTO)
CPICFLAGS = -fPIC
CPPFLAGS = -I/usr/local/opt/gettext/include -I/usr/local/opt/readline/include -I/usr/local/opt/icu4c/include -I/usr/local/include
CXX = clang++ -std=gnu++11
CXXCPP = $(CXX) -E
CXXFLAGS = -I/Library/Java/JavaVirtualMachines/jdk-8.0_172.jdk/Contents/Home/include -I/Library/Java/JavaVirtualMachines/jdk-8.0_172.jdk/Contents/Home/include/darwin $(LTO)
CXXPICFLAGS = -fPIC
CXX98 = clang++
CXX98FLAGS = -I/Library/Java/JavaVirtualMachines/jdk-8.0_172.jdk/Contents/Home/include -I/Library/Java/JavaVirtualMachines/jdk-8.0_172.jdk/Contents/Home/include/darwin $(LTO)
CXX98PICFLAGS = -fPIC
CXX98STD = -std=gnu++98
CXX11 = clang++
CXX11FLAGS = -I/Library/Java/JavaVirtualMachines/jdk-8.0_172.jdk/Contents/Home/include -I/Library/Java/JavaVirtualMachines/jdk-8.0_172.jdk/Contents/Home/include/darwin $(LTO)
CXX11PICFLAGS = -fPIC
CXX11STD = -std=gnu++11
CXX14 = clang++
CXX14FLAGS = -I/Library/Java/JavaVirtualMachines/jdk-8.0_172.jdk/Contents/Home/include -I/Library/Java/JavaVirtualMachines/jdk-8.0_172.jdk/Contents/Home/include/darwin $(LTO)
CXX14PICFLAGS = -fPIC
CXX14STD = -std=gnu++14
CXX17 = clang++
CXX17FLAGS = -I/Library/Java/JavaVirtualMachines/jdk-8.0_172.jdk/Contents/Home/include -I/Library/Java/JavaVirtualMachines/jdk-8.0_172.jdk/Contents/Home/include/darwin $(LTO)
CXX17PICFLAGS = -fPIC
CXX17STD = -std=gnu++17
CXX_VISIBILITY = 
DYLIB_EXT = .dylib
DYLIB_LD = $(CC)
DYLIB_LDFLAGS = -dynamiclib -Wl,-headerpad_max_install_names -undefined dynamic_lookup -single_module -multiply_defined suppress# $(CFLAGS) $(CPICFLAGS) $(LDFLAGS)
DYLIB_LINK = $(DYLIB_LD) $(DYLIB_LDFLAGS) $(LDFLAGS)
ECHO = echo
ECHO_C = \c
ECHO_N = 
ECHO_T = 
F_VISIBILITY = 
## FC is the compiler used for all Fortran as from R 3.6.0
FC = gfortran
FCFLAGS = -g -O2 $(LTO)
## additional libs needed when linking with $(FC), e.g. on some Oracle compilers
FCLIBS_XTRA = 
FFLAGS = -g -O2 $(LTO)
FLIBS =  -L/usr/local/opt/gcc/lib/gcc/9/gcc/x86_64-apple-darwin18/9.2.0 -L/usr/local/opt/gcc/lib/gcc/9 -lgfortran -lquadmath -lm
FPICFLAGS = -fPIC
FOUNDATION_CPPFLAGS = 
FOUNDATION_LIBS = -framework Foundation
JAR = /Library/Java/JavaVirtualMachines/jdk-8.0_172.jdk/Contents/Home/bin/jar
JAVA = /Library/Java/JavaVirtualMachines/jdk-8.0_172.jdk/Contents/Home/jre/bin/java
JAVAC = /Library/Java/JavaVirtualMachines/jdk-8.0_172.jdk/Contents/Home/bin/javac
JAVAH = /Library/Java/JavaVirtualMachines/jdk-8.0_172.jdk/Contents/Home/bin/javah
## JAVA_HOME might be used in the next three.  
## They are for packages 'JavaGD' and 'rJava'
JAVA_HOME = /Library/Java/JavaVirtualMachines/jdk-8.0_172.jdk/Contents/Home
JAVA_CPPFLAGS = -I$(JAVA_HOME)/include -I$(JAVA_HOME)/include/darwin
JAVA_LIBS = -L$(JAVA_HOME)/jre/lib/server -ljvm
JAVA_LD_LIBRARY_PATH = $(JAVA_HOME)/jre/lib/server
LAPACK_LIBS = 
LDFLAGS = -L/usr/local/opt/openblas/lib -L/usr/local/opt/gettext/lib -L/usr/local/opt/readline/lib -L/usr/local/opt/icu4c/lib -L/usr/local/lib
## we only need this is if it is external, as otherwise link to R
LIBINTL= -lintl -Wl,-framework -Wl,CoreFoundation
LIBM = 
LIBR0 = -L"$(R_HOME)/lib$(R_ARCH)"
LIBR1 = -lR
LIBR = -L"$(R_HOME)/lib$(R_ARCH)" -lR
LIBS =  -lpcre -llzma -lbz2 -lz -ldl -lm -liconv -licuuc -licui18n
## needed by R CMD config
LIBnn = lib
LIBTOOL = $(SHELL) "$(R_HOME)/bin/libtool"
LTO = 
## needed to build applications linking to static libR
MAIN_LD = $(CC)
MAIN_LDFLAGS =  $(LDFLAGS)
MAIN_LINK = $(MAIN_LD) $(MAIN_LDFLAGS) $(LDFLAGS)
MKINSTALLDIRS = "$(R_HOME)/bin/mkinstalldirs"
OBJC = clang
OBJCFLAGS = -I/Library/Java/JavaVirtualMachines/jdk-8.0_172.jdk/Contents/Home/include -I/Library/Java/JavaVirtualMachines/jdk-8.0_172.jdk/Contents/Home/include/darwin -fobjc-exceptions $(LTO)
OBJC_LIBS = -lobjc 
OBJCXX = clang++
R_ARCH = 
RANLIB = ranlib
SAFE_FFLAGS = -g -O2 -msse2 -mfpmath=sse
SED = /usr/bin/sed
SHELL = /bin/sh
SHLIB_CFLAGS = 
SHLIB_CXXFLAGS = 
SHLIB_CXXLD = $(CXX)
SHLIB_CXXLDFLAGS = -dynamiclib -Wl,-headerpad_max_install_names -undefined dynamic_lookup -single_module -multiply_defined suppress $(LDFLAGS)
SHLIB_CXX98LD = $(CXX98) $(CXX98STD)
SHLIB_CXX98LDFLAGS = -dynamiclib -Wl,-headerpad_max_install_names -undefined dynamic_lookup -single_module -multiply_defined suppress $(LDFLAGS)
SHLIB_CXX11LD = $(CXX11) $(CXX11STD)
SHLIB_CXX11LDFLAGS = -dynamiclib -Wl,-headerpad_max_install_names -undefined dynamic_lookup -single_module -multiply_defined suppress $(LDFLAGS)
SHLIB_CXX14LD = $(CXX14) $(CXX14STD)
SHLIB_CXX14LDFLAGS = -dynamiclib -Wl,-headerpad_max_install_names -undefined dynamic_lookup -single_module -multiply_defined suppress $(LDFLAGS)
SHLIB_CXX17LD = $(CXX17) $(CXX17STD)
SHLIB_CXX17LDFLAGS = -dynamiclib -Wl,-headerpad_max_install_names -undefined dynamic_lookup -single_module -multiply_defined suppress $(LDFLAGS)
SHLIB_EXT = .so
SHLIB_FFLAGS = 
SHLIB_LD = $(CC)
SHLIB_LDFLAGS = -dynamiclib -Wl,-headerpad_max_install_names -undefined dynamic_lookup -single_module -multiply_defined suppress# $(CFLAGS) $(CPICFLAGS) $(LDFLAGS)
SHLIB_LIBADD = 
## We want to ensure libR is picked up from $(R_HOME)/lib
## before e.g. /usr/local/lib if a version is already installed.
SHLIB_LINK = $(SHLIB_LD) $(SHLIB_LDFLAGS) $(LIBR0) $(LDFLAGS)
SHLIB_OPENMP_CFLAGS = 
SHLIB_OPENMP_CXXFLAGS = 
SHLIB_OPENMP_FFLAGS = 
STRIP_STATIC_LIB = strip -S
STRIP_SHARED_LIB = strip -x
TCLTK_CPPFLAGS = -I/usr/include -I/usr/include 
TCLTK_LIBS = -F/System/Library/Frameworks -framework Tcl -F/System/Library/Frameworks -framework Tk 
YACC = bison -y

## Legacy settings:  no longer used by R as of 3.6.0
## Setting FC often sets F77 (on Solaris make even if set)
## so must follow FC in this file.
F77 = gfortran
FCPICFLAGS = -fPIC
F77_VISIBILITY = 
SHLIB_FCLD = $(FC)
SHLIB_FCLDFLAGS = -dynamiclib -Wl,-headerpad_max_install_names -undefined dynamic_lookup -single_module -multiply_defined suppress $(LDFLAGS)
SHLIB_OPENMP_FCFLAGS = 


## for linking to libR.a
STATIC_LIBR = #  "$(R_HOME)/lib$(R_ARCH)/libR.a"  $(BLAS_LIBS) $(FLIBS)  $(LIBINTL) -lreadline  $(LIBS)

## These are recorded as macros for legacy use in packages
## set on AIX, formerly for old glibc (-D__NO_MATH_INLINES)
R_XTRA_CFLAGS = 
##  was formerly set on HP-UX
R_XTRA_CPPFLAGS =  -I"$(R_INCLUDE_DIR)" -DNDEBUG
## currently unset
R_XTRA_CXXFLAGS = 
## used for gfortran in R > 3.6.0
R_XTRA_FFLAGS = -fno-optimize-sibling-calls

## SHLIB_CFLAGS SHLIB_CXXFLAGS SHLIB_FFLAGS are apparently currently unused
## SHLIB_CXXFLAGS is undocumented, there is no SHLIB_FCFLAGS
ALL_CFLAGS =  $(PKG_CFLAGS) $(CPICFLAGS) $(SHLIB_CFLAGS) $(CFLAGS)
ALL_CPPFLAGS =  -I"$(R_INCLUDE_DIR)" -DNDEBUG $(PKG_CPPFLAGS) $(CLINK_CPPFLAGS) $(CPPFLAGS)
ALL_CXXFLAGS =  $(PKG_CXXFLAGS) $(CXXPICFLAGS) $(SHLIB_CXXFLAGS) $(CXXFLAGS)
ALL_OBJCFLAGS = $(PKG_OBJCFLAGS) $(CPICFLAGS) $(SHLIB_CFLAGS) $(OBJCFLAGS)
ALL_OBJCXXFLAGS = $(PKG_OBJCXXFLAGS) $(CXXPICFLAGS) $(SHLIB_CXXFLAGS) $(OBJCXXFLAGS)
ALL_FFLAGS = -fno-optimize-sibling-calls $(PKG_FFLAGS) $(FPICFLAGS) $(SHLIB_FFLAGS) $(FFLAGS)
## can be overridden by R CMD SHLIB
P_FCFLAGS = $(PKG_FFLAGS)
ALL_FCFLAGS = -fno-optimize-sibling-calls $(P_FCFLAGS) $(FPICFLAGS) $(SHLIB_FFLAGS) $(FCFLAGS)
## LIBR here as a couple of packages use this without SHLIB_LINK
ALL_LIBS = $(PKG_LIBS) $(SHLIB_LIBADD) $(LIBR) $(LIBINTL)

.SUFFIXES:
.SUFFIXES: .c .cc .cpp .d .f .f90 .f95 .m .mm .M .o

.c.o:
	$(CC) $(ALL_CPPFLAGS) $(ALL_CFLAGS) -c $< -o $@
.c.d:
	@echo "making $@ from $<"
	@$(CC) -MM $(ALL_CPPFLAGS) $< > $@
.cc.o:
	$(CXX) $(ALL_CPPFLAGS) $(ALL_CXXFLAGS) -c $< -o $@
.cpp.o:
	$(CXX) $(ALL_CPPFLAGS) $(ALL_CXXFLAGS) -c $< -o $@
.cc.d:
	@echo "making $@ from $<"
	@$(CXX) -M $(ALL_CPPFLAGS) $< > $@
.cpp.d:
	@echo "making $@ from $<"
	@$(CXX) -M $(ALL_CPPFLAGS) $< > $@
.m.o:
	$(OBJC) $(ALL_CPPFLAGS) $(ALL_OBJCFLAGS) -c $< -o $@
.m.d:
	@echo "making $@ from $<"
	@clang -MM $(ALL_CPPFLAGS) $< > $@
.mm.o:
	$(OBJCXX) $(ALL_CPPFLAGS) $(ALL_OBJCXXFLAGS) -c $< -o $@
.M.o:
	$(OBJCXX) $(ALL_CPPFLAGS) $(ALL_OBJCXXFLAGS) -c $< -o $@
.f.o:
	$(FC) $(ALL_FFLAGS) -c $< -o $@
## @FCFLAGS_f9x@ are flags needed to recognise the extensions
.f95.o:
	$(FC) $(ALL_FCFLAGS) -c  $< -o $@
.f90.o:
	$(FC) $(ALL_FCFLAGS) -c  $< -o $@
```

**Comment by randy3k** [comment]
I believe the CFLAGS is set by the Java requirement https://github.com/sethrfore/homebrew-r-srf/blob/master/r.rb#L16 which overrides the `CFLAGS` from its default `-Wall -g -O2`. You may want to file an issue at `homebrew-r-srf` about the missing flag. The missing `-O2` flag may affect the performance of a number of packages.

**Comment by randy3k** [comment]
Fixed with v0.2.2

**Comment by randy3k** [comment]
I found a serious bug in the current version v0.2.2. If you need `Dict()`, please use the master branch for the moment.
